### DISCUSSION TITLE
Get U/V windspeeds over time at a single grid point

### DISCUSSION THREAD

**Discussion by jacob-roth** [blaylockbk/Herbie#79] [Q&A]
Hi! I came across Herbie from HRRR while trying to collect windpseed data at a single location over time. Is this possible to do in Herbie? I'm new to the 2D latlon coordinates referenced [here](https://github.com/blaylockbk/Herbie/discussions/45) and was wondering if there would be an easy way of gathering windspeed data (at a particular `mb` pressure) over time at a single location, e.g., near Minneapolis. (The example referenced in a previous discussion [here](https://github.com/blaylockbk/Herbie/discussions/9#discussioncomment-576417) doesn't show up for me) Thanks!

**Comment by blaylockbk** [comment]
Hi @jacob-roth. Thanks for your question.
Getting the nearest points from a curvilinear grid is not a trivial task. One disadvantage of the GRIB format is that individual points can't be downloaded from the grid--the full grid has to be downloaded, then you can pluck out the point you want. I do have a method in Herbie that I think works pretty well; it's implemented as an xarray custom accessor.

Download a timeseries of Data, only getting the U and V winds at 500 mb.
```python
import pandas as pd
from herbie import Herbie, fast_Herbie_xarray

DATES = pd.date_range('2022-05-25', periods=6, freq='1H')

ds = fast_Herbie_xarray(
    DATES=DATES, 
    searchString='(?:U|V)GRD:500 mb', 
    model='hrrr', 
    product='prs', 
    fxx=0
)
```
<img width="489" alt="image" src="https://user-images.githubusercontent.com/6249613/170407145-70ac2e32-87d5-4191-82bf-01733fef7f8d.png">


Get the nearest point to Minneapolis
```python
minneapolis = (-93.25, 44.98) # (lon, lat)
ds_at_point = ds.herbie.nearest_points(points=minneapolis)
ds_at_point = ds_at_point.squeeze() # I did squeeze because we only requested one point
```
<img width="486" alt="image" src="https://user-images.githubusercontent.com/6249613/170407616-aac1a102-3ac1-46ab-89ec-06d5ec4717e2.png">


Make a plot of the timeseries
```python
ds_at_point.u.plot(label='u', marker='.')
ds_at_point.v.plot(label='v', marker='.')
plt.legend()
```
<img width="298" alt="image" src="https://user-images.githubusercontent.com/6249613/170407641-aba771d3-70b1-4582-b1f5-4c5484fd493a.png">

I plotted the latitude/longituge of the nearest point on a map.
It was (-93.262, 44.969) which is pretty close to the requested point.
<img width="273" alt="image" src="https://user-images.githubusercontent.com/6249613/170408763-35ff1130-10c7-428b-8416-96605bc8759e.png">

**Reply by blaylockbk** [reply]
If you need _**lots**_ of wind data at a point, the Herbie implementation can be inefficient due to the limitation of the GRIB format. The HRRR dataset is available in Zarr format, and that provides a more efficient way to get values at a specific point (https://mesowest.utah.edu/html/hrrr/zarr_documentation/)
